python/cudf/cudf/_lib/copying.pyx

```diff
@@ -608,8 +609,22 @@ def shift(Column input, int offset, object fill_value=None):
     cdef DeviceScalar fill
 
     if isinstance(fill_value, DeviceScalar):
+        if not cudf.utils.dtypes._can_cast(input.dtype, fill_value.dtype):
+            warnings.warn(
+                f"Passing {fill_value.dtype} to shift is deprecated and will "
+                f"raise in a future version"
+                f", pass a {input.dtype} scalar instead.",
```

Comment:
Could you give an example of input/fill value types that will raise this warning?

Reply:
Yes, this is the issue: https://github.com/rapidsai/cudf/issues/10608

Previously pandas supported this:
```python
>>> s = pd.DataFrame()
>>> s['a'] = [1, 2, 3, 4, 5]
>>> s['b'] = pd.Series([10, 11, 12, 13, 14], dtype='datetime64[ns]')
>>> s.groupby('a').shift(1500, fill_value=10)
                              b
0 1970-01-01 00:00:00.000000010
1 1970-01-01 00:00:00.000000010
2 1970-01-01 00:00:00.000000010
3 1970-01-01 00:00:00.000000010
4 1970-01-01 00:00:00.000000010

# dtype is `datetime64[ns]`
```
Now it is:
```python
>>> s.groupby('a').shift(1500, fill_value=10)
    b
0  10
1  10
2  10
3  10
4  10

# dtype is `object`
```


Since this kind of mixed dtype behavior is not allowed in `cudf`, that is when this type of warning is thrown.